docs/product/specs/permissions-revamp.md

```diff
@@ -0,0 +1,153 @@
+# Permissions - Revamp
+
+This spec describes the revamped product considerations for managing Permissions within Mathesar. This is targetted towards the Beta release.
+
+## Goals
+1. Improve our permissions architecture and user flows to utilize PostgreSQL's permission system for DB objects.
```

Comment:
@pavish please note I updated this list's formatting since the combination of the bullets + lettered list just seemed wrong to me somehow.

Reply:
I had the lettered list to be able to reference them later in the docs, like:
> to satisfy Goal (B)

It's only mentioned in one place, I can update the text there.